### PR TITLE
Setup CI related configurations

### PR DESCRIPTION
**Pull request by ts-leojoseph-r**
Created fastlane, Tavis CI and Gemfile to setup CI

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+language: swift
+xcode_workspace: MiniApp.xcworkspace
+xcode_scheme: Tests
+osx_image: xcode10.2
```
Comment (donnie-jp) [inline .travis.yml]: Probably want 11.x?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+before_install:
+- bundle install
+- gem update fastlane cocoapods --no-document
+- pod repo update
+
+script:
+- fastlane ci
+
+deploy:
+  provider: pages
```
Comment (donnie-jp) [inline .travis.yml]: Is GitHub Pages enabled yet?

File: fastlane/Fastfile [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# Import base_config from git
+import_from_git(url: 'https://github.com/rakutentech/ios-buildconfig.git')
+
+default_platform(:ios)
+
+platform :ios do
+  desc "Build All"
+  lane :ci do |options|
+    xcversion(version: "~> 10.2")
```
Comment (donnie-jp) [inline fastlane/Fastfile]: If xcode11 gets set in the `.travis.yml` this will need changed too.

File: Example/MiniApp/AppDelegate.swift [hunk 4]
```diff
@@ -6,9 +6,8 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
     var window: UIWindow?
 
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
-        
+
```
Comment (donnie-jp) [inline Example/MiniApp/AppDelegate.swift]: What are all the whitespace-only changes for?

File: .travis.yml (hunk 1 above)
Comment (ts-leojoseph-r) [reply .travis.yml]: Sure, I will update my xcode as well then.

File: .travis.yml (hunk 2 above)
Comment (ts-leojoseph-r) [reply .travis.yml]: > Is GitHub Pages enabled yet?

No. I couldn't find the settings to enable it. May be I don't have privileges to enable it.

File: Example/MiniApp/AppDelegate.swift (hunk 4 above)
Comment (ts-leojoseph-r) [reply Example/MiniApp/AppDelegate.swift]: By default swiftlint is throwing violation errors for these whitespaces, so I had fixed them. Do you want me to rollback these whitespace changes ?

File: fastlane/Fastfile (hunk 3 above)
Comment (ts-leojoseph-r) [reply fastlane/Fastfile]: Sure, will update the same.

File: .travis.yml (hunk 2 above)
Comment (corycaywood) [reply .travis.yml]: Github pages is automatically enabled when you create a branch named `gh-pages`.

Comment (ts-leojoseph-r) [reply .travis.yml]: Yeah. Got it. Will remove these pages deploy config for now and will add them later once we have created one.

File: .swiftlint.yml [hunk 5]
```diff
@@ -0,0 +1,22 @@
+disabled_rules: # rule identifiers to exclude from running
+  - nesting
+# parameterized rules are first parameterized as a warning level, then error level.
+line_length:
+  - 200 # warning
+  - 300 # error
+function_body_length:
+  - 300
```
Comment (corycaywood) [inline .swiftlint.yml]: It looks like this value is number of lines: https://github.com/realm/SwiftLint/blob/master/Rules.md#L8852

300 lines is a really really long function. I think the Android repo doesn't allow functions longer than 30 lines, but maybe that's a little too low. I'm not sure what the correct length would be for Swift, but how about something like 30 lines for a warning and 50 for an error?

File: Example/MiniApp/AppDelegate.swift (hunk 4 above)
Comment (donnie-jp) [reply Example/MiniApp/AppDelegate.swift]: No you can keep them - if it's from swiftlint that's fine

File: .swiftlint.yml (hunk 5 above)
Comment (donnie-jp) [reply .swiftlint.yml]: Agreed. I think this was needed for the Quick/Nimble BDD tests to get past swiftlint. Maybe it's better to add the Tests folder to the excludes list.

**Comment by donnie-jp** [comment]
> Just one comment from me.
> 
> Also, I've enabled the project on TravisCI. I think it should trigger the next time you push a commit: https://travis-ci.org/rakutentech/ios-miniapp

I had to create the sdkutils job under travis.COM because I think they are forcing all new CI under that domain, both open source and closed e.g. https://travis-ci.com/rakutentech/ios-sdkutils, so maybe there'll be issues trying to use .org.

File: .swiftlint.yml (hunk 5 above)
Comment (ts-leojoseph-r) [reply .swiftlint.yml]: Done

**Comment by corycaywood** [comment]
> > Just one comment from me.
> > Also, I've enabled the project on TravisCI. I think it should trigger the next time you push a commit: https://travis-ci.org/rakutentech/ios-miniapp
> 
> I had to create the sdkutils job under travis.COM because I think they are forcing all new CI under that domain, both open source and closed e.g. https://travis-ci.com/rakutentech/ios-sdkutils, so maybe there'll be issues trying to use .org.

Oh, I didn't know that there is both a travis-ci.org and travis-ci.com. According to [this page](https://docs.travis-ci.com/user/migrate/open-source-on-travis-ci-com), open source projects are being migrated from travis-ci.org to travis-ci.com.

I tried adding the integration for travis-ci.com, but the repo still isn't showing up on .com. @donnie-jp Do you see the option?

**Comment by ts-leojoseph-r** [comment]
> > > Just one comment from me.
> > > Also, I've enabled the project on TravisCI. I think it should trigger the next time you push a commit: https://travis-ci.org/rakutentech/ios-miniapp
> > 
> > 
> > I had to create the sdkutils job under travis.COM because I think they are forcing all new CI under that domain, both open source and closed e.g. https://travis-ci.com/rakutentech/ios-sdkutils, so maybe there'll be issues trying to use .org.
> 
> Oh, I didn't know that there is both a travis-ci.org and travis-ci.com. According to [this page](https://docs.travis-ci.com/user/migrate/open-source-on-travis-ci-com), open source projects are being migrated from travis-ci.org to travis-ci.com.
> 
> I tried adding the integration for travis-ci.com, but the repo still isn't showing up on .com. @donnie-jp Do you see the option?

Yeah, I read about it earlier this week that travis-ci.com changes now. (https://blog.travis-ci.com/2018-05-02-open-source-projects-on-travis-ci-com-with-github-apps)
I did try them in the personal repo with sample code for the CI and the travis ci job created automatically in travis-ci.com. We need to enable integration in Repository settings-> Application

**Comment by donnie-jp** [comment]
> > > Just one comment from me.
> > > Also, I've enabled the project on TravisCI. I think it should trigger the next time you push a commit: https://travis-ci.org/rakutentech/ios-miniapp
> > 
> > 
> > I had to create the sdkutils job under travis.COM because I think they are forcing all new CI under that domain, both open source and closed e.g. https://travis-ci.com/rakutentech/ios-sdkutils, so maybe there'll be issues trying to use .org.
> 
> Oh, I didn't know that there is both a travis-ci.org and travis-ci.com. According to [this page](https://docs.travis-ci.com/user/migrate/open-source-on-travis-ci-com), open source projects are being migrated from travis-ci.org to travis-ci.com.
> 
> I tried adding the integration for travis-ci.com, but the repo still isn't showing up on .com. @donnie-jp Do you see the option?

Seems to be active already https://travis-ci.com/rakutentech/ios-miniapp

**Comment by corycaywood** [comment]
> > > > Just one comment from me.
> > > > Also, I've enabled the project on TravisCI. I think it should trigger the next time you push a commit: https://travis-ci.org/rakutentech/ios-miniapp
> > > 
> > > 
> > > I had to create the sdkutils job under travis.COM because I think they are forcing all new CI under that domain, both open source and closed e.g. https://travis-ci.com/rakutentech/ios-sdkutils, so maybe there'll be issues trying to use .org.
> > 
> > 
> > Oh, I didn't know that there is both a travis-ci.org and travis-ci.com. According to [this page](https://docs.travis-ci.com/user/migrate/open-source-on-travis-ci-com), open source projects are being migrated from travis-ci.org to travis-ci.com.
> > I tried adding the integration for travis-ci.com, but the repo still isn't showing up on .com. @donnie-jp Do you see the option?
> 
> Seems to be active already https://travis-ci.com/rakutentech/ios-miniapp

Weird, it's not showing up for me on the repository listing page. Maybe it's because there aren't any builds yet.

**Pull request merged**